### ISSUE TITLE
Sugestão de melhoria nos useEffect

### ISSUE THREAD

**Issue by thayannevls** [lucasalustiano/omegaHLulTV#12]
`useEffect`s  como [esse](https://github.com/lucasalustiano/omegaHLulTV/blob/main/packages/client/src/components/TeamsCardList/TeamsCardList.jsx#L20), que estão sem dependência e também fazem mais de um comportamento, pode ocasionar bugs e loops, talvez até diminuir um pouco a performance do app.

Pra useEffects como esse, seria legal fazer algo desse tipo:

```jsx
 useEffect(() => {
    if (reverse) {
      handleReverseTeamsList();
    }
  }, [reverse]);

 useEffect(() => {
    if (teamsFiltereds) {
      handleReverseTeamsList();
    }
  }, [teamsFiltereds]);
```

**Comment by lucasalustiano** [comment]
Opa, bom dia, eu entendi o seu ponto só não entendi o exemplo que você usou. Teria como tu explicar melhor o exemplo?

**Comment by thayannevls** [comment]
Opa, acabei colocando o exemplo errado, acho que agora ficou mais claro, dá uma olhada

**Comment by thayannevls** [comment]
Uma outra sugestão complementando com a #13 , é que essa chamada `handleReverseTeamsList` poderia ser feita no contexto. Não sei se tem algum motivo específico para se fazer aqui, mas me parece o contexto deveria saber lidar com as mudanças nos filtros e no reverse que ocasionam no recarregamento a lista de teams, e só repassar para os components filhos a lista atualizada

**Comment by lucasalustiano** [comment]
Concordo, vou dá uma olhada assim que eu tiver tempo, e aviso aqui quando estiver pronto.

**Comment by lucasalustiano** [comment]
Oi @thayannevls, eu dei uma alterada nos useEffects. Quando você tiver tempo, pode dá uma olhada e vê se era aquilo mesmo?

**Comment by thayannevls** [comment]
Boa @lucasalustiano ! Ficou bem melhor :D 🚀
